app/controllers/oidc_controller.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# OidcController is used for performing an OpenID Connect auth
+# flow against Dex, allowing Velum to generate a complete kubeconfig
+# file for a user, including their JWT token
+class OidcController < ApplicationController
+  # allow anyone to start this, the auth will happen inside
+  # Dex.
+  skip_before_action :authenticate_user!  
+
+  def start
+    redirect_to authorization_uri(SecureRandom.hex(16))
+  end
+
+  def done
+    render "start.html.erb"
+  end
```

Comment:
everything below should be in a model or a library, there is currently too much business logic in the controller